cl/beacon/handler/duties_attester.go

```diff
@@ -22,6 +22,25 @@ type attesterDutyResponse struct {
 	Slot                    uint64            `json:"slot,string"`
 }
 
+func (a *ApiHandler) getDependentRoot(s *state.CachingBeaconState, epoch uint64) libcommon.Hash {
+	dependentRootSlot := ((epoch - 1) * a.beaconChainCfg.SlotsPerEpoch) - 3
+	maxIterations := 2048
+	for i := 0; i < maxIterations; i++ {
+		if dependentRootSlot > epoch*a.beaconChainCfg.SlotsPerEpoch {
+			dependentRootSlot = 0
+		}
+
+		dependentRoot, err := s.GetBlockRootAtSlot(dependentRootSlot)
+		if err != nil {
+			dependentRootSlot--
+			continue
+		}
+		fmt.Println(dependentRoot)
+		return dependentRoot
```

Comment:
i feel cannot get the behavior of this function. Is the following code snippet equivalent to your code?

```go
startSlot := ((epoch - 1) * a.beaconChainCfg.SlotsPerEpoch) - 3
tryCount := 0
for slot:=startSlot; slot >=0 && tryCount<2048; slot-- {
  root, err := s.GetBlockRootAtSlot(slot)
  if err == nil {
    return root
  }
  tryCount++
}
return libcommon.Hash{}
```

Reply:
yes, try until works. maybe we miss some blocks